coap/nats/messages.go

```diff
@@ -0,0 +1,27 @@
+// Package nats contains NATS-specific message repository implementation.
+package nats
+
+import (
+	"encoding/json"
+
+	"github.com/mainflux/mainflux/writer"
+	broker "github.com/nats-io/go-nats"
+)
+
+const topic string = "msg.coap"
+
+// Stored NATS connection
+var snc *broker.Conn
```

Comment:
This way of initializing nats publisher is inconsistent with the rest of the platform.